projects/extension/src/background/ConnectionManager.ts

```diff
@@ -215,6 +217,18 @@ export class ConnectionManager
     this.#client = undefined
   }
 
+  runAlarm(): void {
+    for (const [key] of relayChains.entries()) {
+      const network = this.#networks.filter((n) => n.name === key)
+      network[0] &&
+        network[0].chain.databaseContent().then((db) => {
+          chrome.storage.local
+            .set({ [key]: db })
+            .catch((e: Error) => console.error(e))
+        })
+    }
+  }
```

Comment:
There are a couple of things here that I think should be improved:
- It makes for sense to iterate `this.#networks` just once and then only grab those ones that are present in `relayChains`
- the current implementation only catches the error of `chrome.storage.local.set`, the error of the `chain.databaseContent` is being ignored, I think that it makes more sense to take into account both promises when catching.
This suggestion addresses those 2 things:

```suggestion
  runAlarm(): void {
    this.#networks
      .filter(n => relayChains.has(n.name))
      .forEach(async (network) => {
        try {
          const db = await network.chain.databaseContent()
          await chrome.storage.local.set({ [key]: db })
        } catch (e: Error) {
          console.error(e)
        }
      })
    }
```

Reply:
Thank you for the feedback. 
A) the database content should run only for the 4 relay chains that are hard added to extension for speed purposes
B) PR is still draft - ill add the catch to the rest once I manage to resolve a strange test erroe

Reply:
> A) the database content should run only for the 4 relay chains that are hard added to extension for speed purposes

Which is wha the suggsted code is doing in a more efficient way than the previous one.